src/EventTeachingService.cpp

```diff
@@ -422,19 +422,19 @@ void EventTeachingService::handleLearnEvent(const VlcbMessage *msg, unsigned int
     byte index = module_config->findExistingEventByEv(evnum, evval);
     if (index < module_config->EE_MAX_EVENTS)
     {
-      module_config->writeEvent(index, &msg->data[1]);
-      // no need to write eventEV as, by definition, it hasn't changed
-      // recreate event hash table entry
-      module_config->updateEvHashEntry(index);
-      
-      // respond with WRACK
-      controller->sendWRACK();  // Deprecated in favour of GRSP_OK
-      // DEBUG_SERIAL <<F("ets> WRACK sent") << endl;
-
-      // Note that the op-code spec only lists WRACK as successful response.
-      controller->sendGRSP(OPC_EVLRN, getServiceID(), GRSP_OK);
-      ++diagEventsTaught;
-      return;
+      // Ensure that we won't create multiple events for the evval
+      // Search for this NN, EN
+      byte indexNE = module_config->findExistingEvent(nn, en);
+      if (indexNE != index)
```

Comment:
I checked out this PR to run the unit tests. There is a failure here when updating NN/EN for an existing event. It fails where the new NN/EN is not found. This means that the indexNE here is set to  EE_MAX_EVENTS. This is different from the index above (that I want to rename to indexEV1) which is 0, i.e. the first event. 
Suggest changing this condition to:
`   if (indexNE != module_config->EE_MAX_EVENTS || indexNE != index)`
However this is not enough for this case. Now we come to line 446 where we check the index from findExistingEvent() (with the new NN/EN). There is no such event and we search for a free slot with findEventSpace(). This succeeds and we create a new event at index==1. This means that first of all we don't update the event we intended. Secondly we have a second event with the same EV1 value but with a different NN/EN.
So the logic here seems to be a lot more complex.

Reply:
You're right about the condition, but it no longer works properly on my test rig.  It did this morning.  I need to unwind the changes and see where it breaks.  May take until tomorrow.